### ISSUE TITLE
Incidence analysis numbers change if target/outcome cohorts re-used in a different incidence analysis

### ISSUE THREAD

**Issue by altosaar** [OHDSI/Atlas#2647]
### Expected behavior
Incidence analysis should not change if the target or outcome cohorts are used in a second, separate incidence analysis. 

### Actual behavior
The incidence changed from 135 out of 7,995 cases to 140 out of 8,006 cases, *without* clicking on "generate" in the incidence analysis. This should never happen. It causes users to lose work (generation takes several hours, up to a day in this case to return to the original number of 135 out of 7,995).

![image](https://user-images.githubusercontent.com/5317244/146223117-7a904e0a-6b5d-42ce-8a84-2c88eea265c3.png)

(prior to using the target/outcome cohorts in a different incidence analysis, these numbers were 135 and 7,995, and generate was *not* clicked)

### Steps to reproduce behavior
* Create an incidence analysis (E.g.  https://discovery.dbmi.columbia.edu/atlas/#/iranalysis/35/generation)
* Run the incidence analysis
* Duplicate the incidence analysis, and re-use the target and outcome cohorts in a second incidence analysis (E.g. having a different time at risk window); in this case https://discovery.dbmi.columbia.edu/atlas/#/iranalysis/38/generation
* Go back to the first incidence analysis, and the numbers have now changed

cc @t-abdul-basser @noemieelhadad

**Comment by chrisknoll** [comment]
I can't reproduce this behavior:  I created an IR with 1 T and one O. Ran the IR, then made a copy and changed the TAR (in this case, from 30d to 365d).  My results did not get overwritten:

![image](https://user-images.githubusercontent.com/6818777/148786076-4508701e-9279-480e-9b1e-7c9b4a53cc24.png)

and:

![image](https://user-images.githubusercontent.com/6818777/148786146-37afc87f-1977-4cdf-86ef-850c4cf84984.png)

I don't doubt you are seeing something in your environment, but I also checked the code, and the results are written to the database based on the ID of the incidence rate analysis. So I'm not sure how running an IR `123` would be able to change the results in `456`.
